src/Interpreters/InterpreterDropQuery.cpp

```diff
@@ -187,6 +190,11 @@ BlockIO InterpreterDropQuery::executeToTableImpl(const ASTDropQuery & query, Dat
 
             table->shutdown();
 
+#if USE_LIBPQXX
+            if (table->getName() == "PostgreSQLReplica")
+                table->as<StoragePostgreSQLReplica>()->shutdownFinal();
```

Comment:
`shutdown` is called for both detach and drop table. This method should only be called on drop table

Reply:
We have virtual `IStorage::drop()` for this. Is it possible to use it?

Reply:
I used it at first, but it turned out to be not a suitable place to drop nested table there, it did not work there.

Reply:
Then please add a comment with explanation why it's not possible to use `IStorage::drop()`. However, we have similar nested tables in `StorageMaterializedView` and it works somehow, so I hope we can make it work for `PostgreSQLReplica` too.

Reply:
It does not work inside storages's `drop()` method, because it can only be called from database's `drop()` method, but at this point it is not possible to execute InterpreterDropQuery in that database: ` DB::Exception: Database test_database is currently dropped or renamed.`

Reply:
There is the second option - to drop nested tables from `stopReplication()` method instaed of database's `drop` method. I should better do this way. But it anyway requires to be added into InterpreterDropQuery as it was before.

Reply:
>  Is it possible to use it?

There will be a deadlock.

Reply:
It is ok now, I made it the same as for storageMaterializedView.